dmoj/judge.py

```diff
@@ -218,13 +235,29 @@ def _ipc_result(self, report, batch_number: Optional[int], case_number: int, res
                 colored_feedback,
                 colored_aux_codes,
             )
+
         case_padding = '  ' if batch_number is not None else ''
-        report(ansi_style('%sTest case %2d %-3s %s' % (case_padding, case_number, colored_codes[0], case_info)))
+        if self.in_pretests:
+            self.pretest_case += 1
+            report(
+                ansi_style(
+                    '%sPretest case %2d %-3s %s' % (case_padding, self.pretest_case, colored_codes[0], case_info)
+                )
+            )
+        else:
+            self.main_case += 1
+            report(ansi_style('%sTest case %2d %-3s %s' % (case_padding, self.main_case, colored_codes[0], case_info)))
+
         self.packet_manager.test_case_status_packet(case_number, result)
 
-    def _ipc_batch_begin(self, report, batch_number: int) -> None:
+    def _ipc_batch_begin(self, report, _batch_number: int) -> None:
         self.packet_manager.batch_begin_packet()
-        report(ansi_style('#ansi[Batch #%d](yellow|bold)' % batch_number))
+        if self.in_pretests:
+            self.pretest_batch += 1
+            report(ansi_style('#ansi[Pretest Batch #%d](yellow|bold)' % self.pretest_batch))
```

Comment:
Nit to match the rest of the casing

```suggestion
            report(ansi_style('#ansi[Pretest batch #%d](yellow|bold)' % self.pretest_batch))
```